test_agent/cpp/src/TestAgent.h

```diff
@@ -0,0 +1,205 @@
+
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _TEST_AGENT_H_
+#define _TEST_AGENT_H_
+
+#include <Constants.h>
+#include <SocketUTransport.h>
+#include <arpa/inet.h>
+#include <google/protobuf/any.pb.h>
+#include <google/protobuf/util/message_differencer.h>
+#include <netinet/in.h>
+#include <spdlog/spdlog.h>
+#include <unistd.h>
+#include <up-client-zenoh-cpp/client/upZenohClient.h>
+#include <up-core-api/umessage.pb.h>
+#include <up-core-api/uri.pb.h>
+#include <up-cpp/transport/UTransport.h>
+#include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <up-cpp/uuid/factory/Uuidv8Factory.h>
+#include "ProtoConverter.h"
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+#include <iostream>
+#include <map>
+#include <optional>
+#include <string>
+#include <thread>
+#include <variant>
+
+using namespace google::protobuf;
+using namespace std;
+using namespace rapidjson;
+
+using FunctionType = std::variant<std::function<UStatus(Document &)>, std::function<void(Document &)>>;
+
+/**
+ * @class TestAgent
+ * @brief Represents a test agent that communicates with a test manager.
+ *
+ * The TestAgent class is responsible for connecting to a test manager, sending
+ * and receiving messages, and handling various commands. It inherits from the
+ * UListener class to handle incoming messages.
+ */
+class TestAgent : public uprotocol::utransport::UListener {
+public:
+	/**
+	 * @brief Constructs a TestAgent object with the specified transport type.
+	 * @param[in] transportType The type of transport to be used by the agent.
+	 */
```

Comment:
Use C++ style doxygen blocks
```suggestion
	/// @brief Constructs a TestAgent object with the specified transport type.
	/// @param[in] transportType The type of transport to be used by the agent.
```